staging/src/k8s.io/apiserver/plugin/pkg/audit/webhook/webhook.go

```diff
@@ -110,267 +73,33 @@ func loadWebhook(configFile string, groupVersion schema.GroupVersion) (*webhook.
 		[]schema.GroupVersion{groupVersion}, defaultInitialBackoff)
 }
 
-func newBlockingWebhook(configFile string, groupVersion schema.GroupVersion) (*blockingBackend, error) {
-	w, err := loadWebhook(configFile, groupVersion)
-	if err != nil {
-		return nil, err
-	}
-	return &blockingBackend{w}, nil
-}
-
-type blockingBackend struct {
+type backend struct {
 	w *webhook.GenericWebhook
 }
 
-func (b *blockingBackend) Run(stopCh <-chan struct{}) error {
+func (b *backend) Run(stopCh <-chan struct{}) error {
 	return nil
 }
 
-func (b *blockingBackend) Shutdown() {
+func (b *backend) Shutdown() {
 	// nothing to do here
 }
 
-func (b *blockingBackend) ProcessEvents(ev ...*auditinternal.Event) {
+func (b *backend) ProcessEvents(ev ...*auditinternal.Event) {
 	if err := b.processEvents(ev...); err != nil {
 		audit.HandlePluginError(pluginName, err, ev...)
 	}
 }
 
-func (b *blockingBackend) processEvents(ev ...*auditinternal.Event) error {
+func (b *backend) processEvents(ev ...*auditinternal.Event) error {
 	var list auditinternal.EventList
 	for _, e := range ev {
 		list.Items = append(list.Items, *e)
 	}
-	// NOTE: No exponential backoff because this is the blocking webhook
-	// mode. Any attempts to retry will block API server requests.
```

Comment:
@ericchiang @sttts with this change the blocking webhook now retries and blocks API server requests while doing so. Is that acceptible, or should this maintain the fail-fast behavior from before?

Reply:
I would imagine that if somebody configures audit backend to by synchronous, one wants to safely deliver all audit events. Having constant audit blackouts because of temporary network problems seems like far from great. Having 0 retries undermines incentive IMO. Setting initial retry delay to a small value and having 10s seconds delays upon networking problem seems like a more reasonable solution.